Acrolinx.Sidebar/AcrolinxSidebar.cs

```diff
@@ -1,698 +1,715 @@
+            /// This information is used to identify your client on the server.
+            /// Version information about this components might be displayed more prominently.
+            /// </summary>
+            MAIN,
+
+            /// <summary>
+            /// Version information about such components are displayed in the about dialog.
+            /// </summary>
+
+            DEFAULT,
+            /// <summary>
+            /// Version information about such components are displayed in the detail section of the about dialog or not at all.
+            /// </summary>
+            DETAIL
+        }
+        public void RegisterClientComponent(Assembly assembly, string humanReadableName, SoftwareComponentCategory category)
+        {
+            Contract.Requires(assembly != null);
+            Contract.Requires(humanReadableName != null);
+            if (assembly != null)
+            {
+                var name = assembly.GetName();
+                RegisterClientComponent(name.Name, humanReadableName, name.Version.ToString(), category);
+            }
+        }
+
+        private bool mainComponentSet = false;
+        public void RegisterClientComponent(string id, string name, string version, SoftwareComponentCategory category)
+        {
+            Contract.Requires(id != null);
+
+            mainComponentSet |= category == SoftwareComponentCategory.MAIN;
+
+            var clientComponents = InitParameters["clientComponents"].Value<JArray>();
+
+            var component = new JObject();
+            component.Add("id", id.ToLower());
+            component.Add("name", name);
+            component.Add("version", version);
+            component.Add("category", category.ToString().ToUpper());
+            if (category != SoftwareComponentCategory.DETAIL)
+            {
+                clientComponents.Add(component);
+            }
+            Logger.AcroLog.Info(name + "\t" + version);
+        }
+
+        public String Check(IDocument document)
+        {
+            acrolinxPlugin.Document = document;
+            Logger.AcroLog.Debug("Content: " + document.Content);
+
+            var code = "new function(){var c = window.external.getContent(); "
+                + "return acrolinxSidebar.checkGlobal(c, {inputFormat:'" + document.Format.ToString().ToUpper() + "', requestDescription:{documentReference: '"
+                + document.Reference.Replace("\\", "\\\\").Replace("'", "\\'").Replace("\n", "").Replace("\r", "") + "'}, selection:{ranges:" + SerializeSelection(document.Selections) + "}})}();";
+
+            dynamic check = Eval(code);
+            string checkId = check.checkId;
+            return checkId;
+        }
+        public void CancelCheck()
+        {
+            var code = "acrolinxSidebar.onGlobalCheckRejected();";
+            Eval(code);
+            Logger.AcroLog.Info("Check canceled by Acrolinx Integration.");
+        }
+
+        /// <summary>

```

Comment:
@pcdeshmukh, I've added the two methods "ShowMessage" and "GetJavaScriptFriendlyString"!

Reply:
What if integration call ShowMessage and sidebar is < 14.28? Will there be any failure?

imo log this show message to file too.

Reply:
There will be a JavaScript error (function not defined), which doesn't crash the program. In Acrolinx for Office JavaScript errors are logged to file system. I've added the glgging to ShowMessage() as well!

Reply:
That sound good to me.